tests/project/test_project_schemas.py

```diff
@@ -72,15 +72,19 @@ def test_nested_fields_update(
 
 
 def test_project_schema_is_updated_correctly_from_dict(
-    native_app_project_instance: ProjectDefinitionV1,
+    native_app_project_instance: ProjectDefinitionV2,
 ):
-    assert native_app_project_instance.native_app.name == "napp_test"
-    assert native_app_project_instance.native_app.package.distribution == "internal"
-    update_dict = {"native_app": {"package": {"distribution": "external"}}}
-
-    native_app_project_instance.update_from_dict(update_dict)
-    assert native_app_project_instance.native_app.name == "napp_test"
-    assert native_app_project_instance.native_app.package.distribution == "external"
+    pkg_model = native_app_project_instance.entities["pkg"]
+    assert pkg_model.manifest == "app/manifest.yml"
+    assert pkg_model.distribution == "internal"
+    assert pkg_model.meta.role == "test_role"
+
+    update_dict = {"distribution": "external", "meta": {"role": "test_role_2"}}
+    pkg_model.update_from_dict(update_dict)
+    assert pkg_model.manifest == "app/manifest.yml"
+    assert pkg_model.distribution == "external"
+    assert pkg_model.artifacts[0].src == "app/*"
+    assert pkg_model.meta.role == "test_role_2"
```

Comment:
`update_from_dict` doesn't work across `entities` since it's a dict and not a field, so I had to change this test to deep update fields on a single entity